src/nix-store/graphml.cc

```diff
@@ -0,0 +1,91 @@
+#include "graphml.hh"
+#include "util.hh"
+#include "store-api.hh"
+#include "derivations.hh"
+
+#include <iostream>
+
+
+using std::cout;
+
+namespace nix {
+
+
+static inline const string & xmlQuote(const string & s)
+{
+    // Luckily, store paths shouldn't contain any character that needs to be
+    // quoted.
+    return s;
+}
+
+
+static string symbolicName(const string & path)
+{
+    string p = baseNameOf(path);
+    return string(p, p.find('-') + 1);
+}
+
+
+static string makeEdge(const string & src, const string & dst)
+{
+    format f = format("  <edge source=\"%1%\" target=\"%2%\"/>\n")
```

Comment:
Maybe `return fmt(..., xmlQuote(src), xmlQuote(dst));`? Saves the need for calling `f.str()`.